.github/workflows/lint_python.yml

```diff
@@ -15,8 +15,7 @@ jobs:
       - run: pip install codespell mypy pytest ruff safety
       - run: ruff check --output-format=github .
       - run: codespell --ignore-words-list="implementor,mimiced,provicers,re-use,THIRDPARTY,assertIn"  # --skip="*.css,*.js,*.lock"
-      - run: pip install -r requirements-test.txt
-      - run: pip install --editable .
+      - run: pip install --editable ".[test]"
```

Comment:
I'm in the camp of not abusing extras for development dependencies. They are public API and exposing `pip install 'oauthlib [test]'` to the end-users is bad.

Either requirement files or dependency groups describe virtualenv state, not core packaging metadata.

Reply:
Yes, it does. `[test]` references an extra that is included as a part of the core packaging metadata. Now that it doesn't leak into the public space, you need to reference it through `--group`: https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-group

This will work as long as pip is modern: https://pip.pypa.io/en/stable/news/#v25-1.